openpgp/packet/signature.go

```diff
@@ -774,14 +841,19 @@ type outputSubpacket struct {
 func (sig *Signature) buildSubpackets() (subpackets []outputSubpacket, err error) {
 	creationTime := make([]byte, 4)
 	binary.BigEndian.PutUint32(creationTime, uint32(sig.CreationTime.Unix()))
-	subpackets = append(subpackets, outputSubpacket{true, creationTimeSubpacket, false, creationTime})
+	subpackets = []outputSubpacket{
+		outputSubpacket{true, creationTimeSubpacket, false, creationTime},
+	}
 
-	if sig.IssuerKeyId != nil {
+	if sig.IssuerKeyId != nil && sig.Version() == 4 {
 		keyId := make([]byte, 8)
 		binary.BigEndian.PutUint64(keyId, *sig.IssuerKeyId)
-		subpackets = append(subpackets, outputSubpacket{true, issuerSubpacket, false, keyId})
+		subpackets = append(subpackets, outputSubpacket{true, issuerSubpacket, true, keyId})
+	}
+	if sig.IssuerKeyFingerprint != nil {
+		contents := append([]uint8{uint8(sig.Version())}, sig.IssuerKeyFingerprint...)
```

Comment:
Technically speaking, this should be the key version, not the signature version. Obviously, they are always the same right now, but I think technically speaking this is not required, so it might be nice to make it obvious in the code, e.g. by adding an explicit `IssuerKeyVersion *int` property?

Reply:
For V5 they MUST be equal and for V4 they SHOULD be equal, this is complying with that. `Signature` has already 35 fields or so, I'd prefer not to add more

Reply:
Yeah, I'm not saying we should change that behavior, I'm just saying we should make it clear what the spec says by making the version come from the key instead of the signature.

You could also pass the signing key to this function, if you want.